### PR TITLE
sockets: add packet sockets

### PR DESCRIPTION
**Pull request by ziemleszcz**
JIRA: RTOS-38

### CODE REVIEW

File: port/netpacket.c [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+		pbuf_dechain(packet);
+		pbuf_free(packet);
+	}
+
+	return ret;
+}
+
+static void netpacket_recv(struct netpacket_pcb *pcb, struct pbuf *p, struct eth_addr *src_addr)
+{
+    struct pbuf *q;
```
Comment (nalajcie) [inline port/netpacket.c:120]: invalid indentation

File: port/netpacket.c [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+	if ((conn != NULL) && sys_mbox_valid(&conn->recvmbox)) {
+#if LWIP_SO_RCVBUF
+		int recv_avail;
+		SYS_ARCH_GET(conn->recv_avail, recv_avail);
+		if ((recv_avail + (int)(p->tot_len)) > conn->recv_bufsize) {
+			return;
+		}
+#endif /* LWIP_SO_RCVBUF */
+		/* copy the whole packet into new pbufs */
+        q = pbuf_clone(PBUF_RAW, PBUF_RAM, p);
```
Comment (nalajcie) [inline port/netpacket.c:133]: indentation

File: port/netpacket.c [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+			continue;
+
+		if ((pcb->protocol != ETHERTYPE_ALL) && (pcb->protocol != type))
+			continue;
+
+		struct eth_addr src_addr;
+		SMEMCPY(&src_addr, netif->hwaddr, sizeof(struct eth_addr));
+
+		if (pcb->type == SOCK_DGRAM) {
+            /* remove ethernet header from pbuf */
```
Comment (nalajcie) [inline port/netpacket.c:222]: indentation

File: port/netpacket.c [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+			pbuf_header(p, -(s16_t)sizeof(struct eth_hdr));
+		}
+
+		netpacket_recv(pcb, p, &src_addr);
+
+		if (pcb->type == SOCK_DGRAM) {
+            /* restore ethernet header to pbuf */
```
Comment (nalajcie) [inline port/netpacket.c:229]: indentation

**Review by nalajcie — COMMENTED** [review]
pointed out some small issues.

Waiting for https://github.com/phoenix-rtos/lwip/pull/3 merge and submodule update for build checks to pass

**Comment by nalajcie** [comment]
FYI, due to Github Actions imitations, not all clang-format issues are shown inline in the code, You may check the builder output log here: https://github.com/phoenix-rtos/phoenix-rtos-lwip/pull/27/checks?check_run_id=2762812662

**Comment by nalajcie** [comment]
please note that there are still some styling errors, see "Files changed" tab

**Comment by nalajcie** [comment]
Please rebase on top of current master (updated lib-lwip submodule), so CI would check if it builds

Or it doesn't compile because of not-yet-merged https://github.com/phoenix-rtos/libphoenix/pull/92 ?

**Comment by ziemleszcz** [comment]
It won't build without https://github.com/phoenix-rtos/libphoenix/pull/92.

**Comment by nalajcie** [comment]
- merged https://github.com/phoenix-rtos/libphoenix/pull/92
 - updating submodules in phoenix-rtos-project: https://github.com/phoenix-rtos/phoenix-rtos-project/pull/95
 - will re-run CI jobs then to veify


I hope the below warnings will disappear, but I think it might not be related... :
image

**Review by nalajcie — APPROVED** [review]
All checks passed, LGTM.

**Pull request merged**